### PR TITLE
Add raw recovery progress to cat recovery API

### LINKED ISSUE #17022: Add bytes and files progress to cat recovery
The cat recovery API does not output fields showing the in-progress bytes and files recovered (it only shows bytes and files percent). These fields should be added to the cat recovery API. Additionally, there is some inconsistency in the naming of the fields. Today we have:

`bytes`: bytes to be recovered
`total_bytes`: total bytes on disk
`bytes_percent`: percent of bytes to be recovered that have been recovered

`files`: files to be recovered
`total_files`: total files on disk
`files_percent`: percent of files to be recovered that have been recovered

`translog`: translog ops recovered
`translog_total`: total translog ops to be recovered
`translog_percent`: percent of translog ops to be recovered that have been recovered.

This means that:

`translog_percent = translog / translog_total`

but that

`bytes_percent != bytes / bytes_total`

and

`files_percent = files / files_total`.

Thus, we should add `bytes_recovered` and `files_recovered`, rename `translog` to `translog_ops_recovered` and rename `translog_total` to `translog_ops` and then it will be

`bytes_percent = bytes_recovered / bytes`,
`files_percent = files_recovered / files`

and

`translog_ops_percent = translog_ops_recovered / translog_ops`.

### PR DESCRIPTION
**Pull request by jasontedor**
This commit adds fields bytes_recovered and files_recovered to the cat
recovery API. These fields, respectively, indicate the total number of
bytes and files recovered. Additionally, for consistency, some translog
recovery fields have been renamed.

Closes #17022

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/rest/action/cat/RestRecoveryAction.java [hunk 1]
```diff
@@ -92,14 +92,16 @@ protected Table getTableWithHeader(RestRequest request) {
                 .addCell("repository", "alias:rep;desc:repository")
                 .addCell("snapshot", "alias:snap;desc:snapshot")
                 .addCell("files", "alias:f;desc:number of files to recover")
+                .addCell("files_recovered", "alias:fr;desc:files recovered")
                 .addCell("files_percent", "alias:fp;desc:percent of files recovered")
-                .addCell("bytes", "alias:b;desc:size to recover in bytes")
-                .addCell("bytes_percent", "alias:bp;desc:percent of bytes recovered")
                 .addCell("total_files", "alias:tf;desc:total number of files")
```
Comment (tlrx) [inline core/src/main/java/org/elasticsearch/rest/action/cat/RestRecoveryAction.java]: Maybe `files_total`? We already break things, so...

File: core/src/main/java/org/elasticsearch/rest/action/cat/RestRecoveryAction.java [hunk 2]
```diff
@@ -92,14 +92,16 @@ protected Table getTableWithHeader(RestRequest request) {
...
+                .addCell("bytes", "alias:b;desc:number of bytes to recover")
+                .addCell("bytes_recovered", "alias:br;desc:bytes recovered")
+                .addCell("bytes_percent", "alias:bp;desc:percent of bytes recovered")
                 .addCell("total_bytes", "alias:tb;desc:total number of bytes")
```
Comment (tlrx) [inline core/src/main/java/org/elasticsearch/rest/action/cat/RestRecoveryAction.java]: Maybe `bytes_total`?

**Comment by tlrx** [comment]
Left minor comment, but LGTM

**Comment by jasontedor** [comment]
> Left minor comment, but LGTM

Thanks @tlrx, I think what you suggest is better, it makes it easier to read in the output. I pushed e4ab131d72a8ae54b00dbe8eadc5105e3f19bed4.

**Pull request closed without merging**